src/Service/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -61,10 +61,12 @@ public string Build(SqlQueryStructure structure)
         /// <inheritdoc />
         public string Build(SqlInsertStructure structure)
         {
-
+            string predicates = JoinPredicateStrings(structure.DbPolicyPredicates);
             return $"INSERT INTO {QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier(structure.DatabaseObject.Name)} ({Build(structure.InsertColumns)}) " +
                     $"OUTPUT {MakeOutputColumns(structure.OutputColumns, OutputQualifier.Inserted)} " +
-                    $"VALUES ({string.Join(", ", structure.Values)});";
+                    $"SELECT {Build(structure.InsertColumns)} " +
+                    $"FROM (VALUES({string.Join(", ", structure.Values)})) T({Build(structure.InsertColumns)}) " +
+                    $"WHERE {predicates};";
```

Comment:
Does this fulfill the restrictions of the database policy? this looks like it just restricts the results, but does a record still get inserted?

Reply:
No, it does fulfill the requirement.